docs/deploying a sample wildfly application.md

```diff
@@ -0,0 +1,144 @@
+# Deploying a sample WildFly application
+
+User Story:
+User downloads and starts Minishift (which deploys an instance of OpenShift Origin locally) in order to launch a sample Hello World application using WildFly.
+
+<!-- MarkdownTOC -->
+
+- [Objective](#objective)
+- [Prerequisites](#prerequisites)
+- [Deploying a Hello World WildFly application](#deploying-a-hello-world-wildfly-application)
+- [Additional resources](#additional resources)
+
+
+<!-- /MarkdownTOC -->
+
+<a name="objective"></a>
+## Objective
+To deploy a Hello World application using WildFly, on an OpenShift Origin instance provisioned by Minishift.
+
+<a name="prerequisites"></a>
+## Prerequisites:
+You can create a new application in OpenShift Origin either from a source code, an image or a template. In this example we use source code along with the WildFly Imagestream to deploy a Hello World application.
+
+1. To use source code to deploy the application, you need to have your source code hosted on the internet and accessible through GitHub. Fork and clone this [Sample Application](https://github.com/openshiftdemos/os-sample-java-web) repository. The Sample Application used here contains the source code for a simple Java web application with a single Jsp file that prints out a Hello World message and is hosted on GitHub.  
+
+1. Ensure that you have installed and started Minishift as per the [Installation documentation](https://github.com/minishift/minishift/blob/master/README.md#installation), this will automatically provision an instance of OpenShift Origin for you.
+
+<a name="deploying-a-hello-world-wildfly-application"></a>
+## Deploying a Hello World WildFly application
+
+Once you have started Minishift, follow these instructions to deploy a Hello World application using WildFly in OpenShift:
+
+1. Login to your OpenShift account, if you are not already logged in:
+
+        $ oc login https://<minishift ip>:8443 -u developer -p developer
+        Login successful.
+
+        You have one project on this server: "myproject"
+
+        Using project "myproject".
+
+1. (Optional step) The current example uses the default project “myproject”. You can either choose to use “myproject” for deployment of this application or create a fresh project for the same as:
+
+        $ oc new-project <projectname> --display-name="Abc_123"
+        Now using project "projectname" on server "https://192.168.42.232:8443".
+
+        You can add applications to this project with the 'new-app' command. For example, try:
+
+        oc new-app centos/ruby-22-centos7~https://github.com/openshift/ruby-ex.git
+
+        to build a new example application in Ruby.
+
+1. To create a WildFly application you need to use either a [WildFly Imagestream or a template](https://docs.openshift.org/latest/install_config/imagestreams_templates.html). Check the available Imagestreams already registered in your OpenShift Origin cluster by doing one of the following:
```

Comment:
The Wildfly image stream provides the Wildfly base images, but it does not give you an app.